src/lib/es5.d.ts

```diff
@@ -1273,7 +1273,9 @@ declare type PropertyDecorator = (target: Object, propertyKey: string | symbol)
 declare type MethodDecorator = <T>(target: Object, propertyKey: string | symbol, descriptor: TypedPropertyDescriptor<T>) => TypedPropertyDescriptor<T> | void;
 declare type ParameterDecorator = (target: Object, propertyKey: string | symbol, parameterIndex: number) => void;
 
-declare type PromiseConstructorLike = new <T>(executor: (resolve: (value?: T | PromiseLike<T>) => void, reject: (reason?: any) => void) => void) => PromiseLike<T>;
+interface PromiseConstructorLike {

```

Comment:
I don't think this specific change is necessary. `PromiseConstructorLike` is only used by the type system to determine if an explicit `Promise`-like return type in a down-level ES5 async function can be used as a `Promise` constructor. Changing the type could adversely affect people using this behavior.

Reply:
Done.